pkg/tso/keyspace_group_manager.go

```diff
@@ -837,18 +837,23 @@ func (kgm *KeyspaceGroupManager) checkTSOSplit(
 	if err != nil {
 		return err
 	}
+	// If the split source TSO is not greater than the newly split TSO, we don't need to do anything.
 	if tsoutil.CompareTimestamp(&splitSourceTSO, &splitTSO) <= 0 {
-		log.Debug("the split source TSO is not greater than the newly split TSO",
+		log.Info("the split source tso is less than the newly split tso",
 			zap.Int64("split-source-tso-physical", splitSourceTSO.Physical),
 			zap.Int64("split-source-tso-logical", splitSourceTSO.Logical),
 			zap.Int64("split-tso-physical", splitTSO.Physical),
-			zap.Int64("split-tso-logical", splitTSO.Logical),
-		)
-		return nil
+			zap.Int64("split-tso-logical", splitTSO.Logical))
+		return kgm.finishSplitKeyspaceGroup(keyspaceGroupID)
 	}
 	// If the split source TSO is greater than the newly split TSO, we need to update the split
 	// TSO to make sure the following TSO will be greater than the split keyspaces ever had
 	// in the past.
+	log.Info("the split source tso is greater than the newly split tso",
```

Comment:
The first log will only print once since it successfully splits. But here might print many times?

Reply:
Moved. PTAL.